src/TraceEvent/Parsers/GCDynamicTraceEventParser.cs

```diff
@@ -556,73 +404,17 @@ public sealed class CommittedUsage
         public long TotalBookkeepingCommitted { get; internal set; }
     }
 
-    public sealed class HeapCountSampleTraceEvent : GCDynamicEvent
+    public sealed class DynamicEvent
     {
-        public short Version { get { return BitConverter.ToInt16(DataField, 0); }}
-        public long GCIndex { get { return BitConverter.ToInt64(DataField, 2); }}
-        public long ElapsedTimeBetweenGCs { get { return BitConverter.ToInt64(DataField, 10); }}
-        public long GCPauseTime { get { return BitConverter.ToInt64(DataField, 18); }}
-        public long MslWaitTime { get { return BitConverter.ToInt64(DataField, 26); }}
-
-        internal override TraceEventID ID => TraceEventID.Illegal - 12;
-        internal override string TaskName => "GC";
-        internal override string OpcodeName => "HeapCountSample";
-        internal override string EventName => "GC/HeapCountSample";
-
-        private string[] _payloadNames;
-
-        internal override string[] PayloadNames
-        {
-            get
-            {
-                if (_payloadNames == null)
-                {
-                    _payloadNames = new string[] { "Version", "GCIndex", "ElapsedTimeBetweenGCs", "GCPauseTime", "MslWaitTime" };
-                }
-
-                return _payloadNames;
-            }
-        }
-
-        internal override object PayloadValue(int index)
+        public DynamicEvent(string name, DateTime timeStamp, byte[] payload)
```

Comment:
Can we rename this to GCDynamicEvent to scope it appropriately?  It would also be good to add some comments around why this object exists vs. just using RawDynamicTraceEvent.  I assume this is to avoid the Clone and finalization/disposal cost.